clients/fides-js/src/components/Overlay.tsx

```diff
@@ -1,73 +1,196 @@
+import { h, FunctionComponent } from "preact";
+import { useEffect, useState, useCallback, useMemo } from "preact/hooks";
 import {
-  PrivacyExperience,
+  ConsentMethod,
   FidesOptions,
+  PrivacyExperience,
   PrivacyNotice,
-  UserGeolocation,
+  SaveConsentPreference,
 } from "../lib/consent-types";
 import ConsentBanner from "./ConsentBanner";
-import { CookieKeyConsent } from "../lib/cookie";
-import ConsentModal from "./ConsentModal";
 
 import { updateConsentPreferences } from "../lib/preferences";
+import {
+  debugLog,
+  hasActionNeededNotices,
+  transformConsentToFidesUserPreference,
+} from "../lib/consent-utils";
+import { FidesCookie } from "../lib/cookie";
+
+import "./fides.css";
+import { useA11yDialog } from "../lib/a11y-dialog";
+import ConsentModal from "./ConsentModal";
 
 export interface OverlayProps {
-  consentDefaults: CookieKeyConsent;
   options: FidesOptions;
   experience: PrivacyExperience;
-  geolocation?: UserGeolocation;
+  cookie: FidesCookie;
+  fidesRegionString: string;
+  modalLinkEl?: HTMLElement | null;
 }
 
-const Overlay: FunctionComponent<OverlayProps> = ({ experience }) => {
-  const [modalIsOpen, setModalIsOpen] = useState(false);
+const Overlay: FunctionComponent<OverlayProps> = ({
+  experience,
+  options,
+  fidesRegionString,
+  cookie,
+  modalLinkEl,
+}) => {
+  const [bannerIsOpen, setBannerIsOpen] = useState(false);
+  const { instance, attributes } = useA11yDialog({
+    id: "fides-modal",
+    role: "dialog",
+    title: experience?.experience_config?.title || "",
+  });
+
+  const handleOpenModal = useCallback(() => {
+    if (instance) {
+      instance.show();
+    }
+  }, [instance]);
+
+  const handleCloseModal = useCallback(() => {
+    if (instance) {
+      instance.hide();
+    }
+  }, [instance]);
+
+  useEffect(() => {
+    const delayBanner = setTimeout(() => {
+      setBannerIsOpen(true);
+    }, 100);
+    return () => clearTimeout(delayBanner);
+  }, [setBannerIsOpen]);
+
+  useEffect(() => {
+    if (modalLinkEl) {
+      debugLog(
+        options.debug,
+        "Modal link element found, updating it to show and trigger modal on click."
+      );
+      // Update modal link to trigger modal on click
+      const modalLink = modalLinkEl;
+      modalLink.onclick = () => {
+        handleOpenModal();
+        setBannerIsOpen(false);
+      };
+      // Update to show the pre-existing modal link in the DOM
+      modalLink.classList.add("fides-modal-link-shown");
+    } else {
+      debugLog(options.debug, "Modal link element not found.");
+    }
+  }, [modalLinkEl, options.debug, handleOpenModal]);
+
+  const showBanner = useMemo(
+    () => experience.show_banner && hasActionNeededNotices(experience),
+    [experience]
+  );
 
   if (!experience.experience_config) {
+    debugLog(options.debug, "No experience config found");
     return null;
   }
 
   const privacyNotices = experience.privacy_notices ?? [];
 
   const onAcceptAll = () => {
```

Comment:
perhaps we can refactor the `onAcceptAll`, `onRejectAll`, and `onSavePreferences` to be more streamlined?